### PR TITLE
fix: don't override user defined MaxListeners

### PR DESCRIPTION
**Pull request by fawazahmed0**
Fixes https://github.com/nodejs/node/issues/47449

Please note, I haven't tested this PR.

## Status

- [x] I have read and agreed to the [Developer's Certificate of Origin][cert]
- [ ] Tested
- [ ] Benchmarked (**optional**)
- [ ] Documented
- [ ] Review ready
- [ ] In review
- [ ] Merge ready

[cert]: https://github.com/nodejs/undici/blob/main/CONTRIBUTING.md

### CODE REVIEW

**Comment by KhafraDev** [comment]
Please add a test

File: test/fetch/max-listeners.js [hunk 1]
```diff
@@ -0,0 +1,13 @@
+const { setMaxListeners, getMaxListeners } = require('events')
```
Comment (tsctx) [inline test/fetch/max-listeners.js:1]:
```suggestion
'use strict'

const { setMaxListeners, getMaxListeners } = require('events')
```

File: test/fetch/max-listeners.js [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+const { test } = require('node:test')
+const assert = require('node:assert')
+const { fetch } = require('../..')
+
+test('test max listeners', async (t) => {
+    const controller = new AbortController();
+    setMaxListeners(Infinity, controller.signal)
+    let url = URL.createObjectURL(new Blob())
+    for(let i=0;i<1600;i++)
+        fetch(url, { signal: controller.signal })
```
Comment (tsctx) [inline test/fetch/max-listeners.js:11]:
```suggestion
    new Request('http://asd', { signal: controller.signal })
```

File: test/fetch/max-listeners.js [hunk 3]
```diff
@@ -0,0 +1,13 @@
+const { setMaxListeners, getMaxListeners } = require('events')
+const { test } = require('node:test')
+const assert = require('node:assert')
+const { fetch } = require('../..')
+
+test('test max listeners', async (t) => {
```
Comment (tsctx) [inline test/fetch/max-listeners.js:6]:
```suggestion
test('test max listeners', (t) => {
```

File: test/fetch/max-listeners.js [hunk 4]
```diff
@@ -0,0 +1,13 @@
+const { setMaxListeners, getMaxListeners } = require('events')
+const { test } = require('node:test')
+const assert = require('node:assert')
+const { fetch } = require('../..')
```
Comment (tsctx) [inline test/fetch/max-listeners.js:4]:
```suggestion
const { Request } = require('../..')
```

Comment (tsctx) [reply test/fetch/max-listeners.js:6]: Yes.

**Comment by tsctx** [comment]
it seems to lint error

File: test/fetch/max-listeners.js [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+
+const { setMaxListeners, getMaxListeners, defaultMaxListeners } = require('events')
+const { test } = require('node:test')
+const assert = require('node:assert')
+const { Request } = require('../..')
+
+test('test max listeners', (t) => {
+  const controller = new AbortController()
+  setMaxListeners(Infinity, controller.signal)
+  for (let i = 0; i <= defaultMaxListeners; i++) { new Request('http://asd', { signal: controller.signal }) }
```
Comment (tsctx) [inline test/fetch/max-listeners.js:11]:
```suggestion
  for (let i = 0; i <= defaultMaxListeners; i++) {
    // eslint-disable-next-line no-new
    new Request('http://asd', { signal: controller.signal })
  }
```

**Review by tsctx — COMMENTED** [review]
lgtm

**Review by Ethan-Arrowood — APPROVED** [review]

**Review by KhafraDev — APPROVED** [review]

**Pull request merged**